swagger/formats.go

```diff
@@ -0,0 +1,35 @@
+package swagger
+
+import (
+	"regexp"
+
+	"github.com/go-openapi/strfmt"
+)
+
+func InitCustomFormats() {
+	m := mongoID("")
+	strfmt.Default.Add("mongo-id", &m, isMongoID)
+}
+
+var mongoRegExp = regexp.MustCompile("^[0-9a-f]{24}$")
+
+func isMongoID(str string) bool {
+	return mongoRegExp.MatchString(str)
```

Comment:
i thought about using a `bson.ObjectId` type and was torn. It makes things a bit easier for clients / servers, though maybe not that much easier. It also ties us more tightly to Mongo and I was a bit wary of making that even harder to change.

I definitely think this is worth re-visiting, but I'm not ready to commit yet. Let me know if you strongly disagree.